apps/client/src/features/modals/sheets-modal/SheetsModal.tsx

```diff
@@ -0,0 +1,314 @@
+      return;
+    }
+
+    const selectedFile = event.target.files[0];
+    uploadSheetClientFile(selectedFile)
+      .then(() => {
+        setErrors({ ...errors, clientSecret: '' });
+      })
+      .catch((err) => {
+        if (err.response.data.message) {
+          setErrors({ ...errors, clientSecret: err.response.data.message });
+        }
+      });
+    updateSheetState();
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      updateSheetState();
+    }
+  }, [isOpen]);
+
+  const handleAuthenticate = () => {
+    getSheetsAuthUrl()
+      .then((data) => {
+        openLink(data);
+        window.addEventListener('focus', () => updateSheetState(), { once: true });
+      })
+      .catch((err) => {
+        const message = maybeAxiosError(err);
+        setErrors({ ...errors, authenticate: message });
+      });
+  };
+
+  const handlePullData = () => {
+    postPreviewSheet(sheetRef.current?.value ?? '', worksheetRef.current?.value ?? '').then((data) => {
+      setProject(data.project);
+      setRundown(data.rundown);
+      setUserFields(data.userFields);
+    });
+  };
+
+  const handlePushData = () => {
+    const currentSheetId = sheetRef.current?.value ?? '';
+    const currentWorksheet = worksheetRef.current?.value ?? '';
+    postPushSheet(currentSheetId, currentWorksheet);
+  };
+
+  const handleFinalise = async () => {
+    // this step is currently only used for excel files, after preview
+    if (rundown && userFields && project) {
+      let doClose = false;
+      try {
+        await patchData({ rundown, userFields, project });
+        queryClient.setQueryData(RUNDOWN, rundown);
+        queryClient.setQueryData(USERFIELDS, userFields);
+        queryClient.setQueryData(PROJECT_DATA, project);
+        await queryClient.invalidateQueries({
+          queryKey: [...RUNDOWN, ...USERFIELDS, ...PROJECT_DATA],
+        });
+        doClose = true;
+      } catch (error) {
+        const message = maybeAxiosError(error);
+        console.log(message);
+        // setErrors(`Failed applying changes ${message}`);
+      } finally {
+        if (doClose) {
+          handleClose();
+        }
+      }
+    }
+  };
+
+  return (
+    <Modal
+      onClose={handleClose}
+      isOpen={isOpen}
+      closeOnOverlayClick={false}
+      motionPreset='slideInBottom'
+      size='xl'
+      scrollBehavior='inside'
+      preserveScrollBarGap
+      variant='ontime-upload'
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Rundown from sheets</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <Alert status='info' variant='ontime-on-light-info'>
+            <AlertIcon />
+            <div style={{ display: 'flex', flexDirection: 'column' }}>
+              <AlertTitle>Sync with Google Sheets</AlertTitle>
+              <AlertDescription>
+                <ModalLink href='our-docs'>For more information, see the docs</ModalLink>
```

Comment:
here is the link
https://ontime.gitbook.io/v2/features/google-sheet-experimental